reference/test/matrix/dense_kernels.cpp

```diff
@@ -295,6 +296,130 @@ TEST_F(Dense, MovesToCsr)
 }
 
 
+TEST_F(Dense, ConvertsToEll)
+{
+    auto ell_mtx = gko::matrix::Ell<>::create(mtx4->get_executor());
+
+    mtx4->convert_to(ell_mtx.get());
+
+    auto v = ell_mtx->get_const_values();
+    auto c = ell_mtx->get_const_col_idxs();
+
+    ASSERT_EQ(ell_mtx->get_num_rows(), 2);
+    ASSERT_EQ(ell_mtx->get_num_cols(), 3);
+    ASSERT_EQ(ell_mtx->get_max_nonzeros_per_row(), 3);
+    EXPECT_EQ(c[0], 0);
+    EXPECT_EQ(c[1], 1);
+    EXPECT_EQ(c[2], 1);
+    EXPECT_EQ(c[3], 0);
+    EXPECT_EQ(c[4], 2);
+    EXPECT_EQ(c[5], 0);
+    EXPECT_EQ(v[0], 1.0);
+    EXPECT_EQ(v[1], 5.0);
+    EXPECT_EQ(v[2], 3.0);
+    EXPECT_EQ(v[3], 0.0);
+    EXPECT_EQ(v[4], 2.0);
+    EXPECT_EQ(v[5], 0.0);
+}
+
+
+TEST_F(Dense, MovesToEll)
+{
+    auto ell_mtx = gko::matrix::Ell<>::create(mtx4->get_executor());
+
+    mtx4->move_to(ell_mtx.get());
+
+    auto v = ell_mtx->get_const_values();
+    auto c = ell_mtx->get_const_col_idxs();
+
+    ASSERT_EQ(ell_mtx->get_num_rows(), 2);
+    ASSERT_EQ(ell_mtx->get_num_cols(), 3);
+    ASSERT_EQ(ell_mtx->get_max_nonzeros_per_row(), 3);
+    EXPECT_EQ(c[0], 0);
+    EXPECT_EQ(c[1], 1);
+    EXPECT_EQ(c[2], 1);
+    EXPECT_EQ(c[3], 0);
+    EXPECT_EQ(c[4], 2);
+    EXPECT_EQ(c[5], 0);
+    EXPECT_EQ(v[0], 1.0);
+    EXPECT_EQ(v[1], 5.0);
+    EXPECT_EQ(v[2], 3.0);
+    EXPECT_EQ(v[3], 0.0);
+    EXPECT_EQ(v[4], 2.0);
+    EXPECT_EQ(v[5], 0.0);
+}
+
+
+TEST_F(Dense, ConvertsToEllWithPadding)
+{
+    auto ell_mtx = gko::matrix::Ell<>::create(mtx4->get_executor());
+
+    mtx4->convert_to(ell_mtx.get(), 16);
+
+    auto max_nnz_per_row = ell_mtx->get_max_nonzeros_per_row();
+    auto padding = ell_mtx->get_padding();
+    ASSERT_EQ(ell_mtx->get_num_rows(), 2);
+    ASSERT_EQ(ell_mtx->get_num_cols(), 3);
+    ASSERT_EQ(max_nnz_per_row, 3);
+    ASSERT_EQ(padding, 16);
+    for (gko::size_type col_idx = 0; col_idx < max_nnz_per_row; col_idx++) {
+        for (gko::size_type row = 0; row < padding; row++) {
+            if (col_idx == 0 && row == 0) {
+                EXPECT_EQ(ell_mtx->col_at(row, col_idx), 0);
+                EXPECT_EQ(ell_mtx->val_at(row, col_idx), 1.0);
+            } else if (col_idx == 0 && row == 1) {
+                EXPECT_EQ(ell_mtx->col_at(row, col_idx), 1);
+                EXPECT_EQ(ell_mtx->val_at(row, col_idx), 5.0);
+            } else if (col_idx == 1 && row == 0) {
+                EXPECT_EQ(ell_mtx->col_at(row, col_idx), 1);
+                EXPECT_EQ(ell_mtx->val_at(row, col_idx), 3.0);
+            } else if (col_idx == 2 && row == 0) {
+                EXPECT_EQ(ell_mtx->col_at(row, col_idx), 2);
+                EXPECT_EQ(ell_mtx->val_at(row, col_idx), 2.0);
+            } else {
+                EXPECT_EQ(ell_mtx->col_at(row, col_idx), 0);
+                EXPECT_EQ(ell_mtx->val_at(row, col_idx), 0.0);
+            }
+        }
+    }
```

Comment:
You should use the [KISS principle](http://principles-wiki.net/principles:keep_it_simple_stupid) when writing unit tests. The goal is to have them as readable as possible.
In this case I would do the following:

Use the smallest possible matrix which can exercise the feature being tested, in this case, maybe a 2-by-4 dense matrix with at most 3 non-zeros per row, and set the padding to 3. Then, I would just explicitly check all 9 values of the result, without writing any for loops in the test.

Reply:
Fix it